### PR TITLE
Allow jitter boolean to be set through `nuts_sampler_kwargs`

### PR DESCRIPTION
**Pull request by VMBoehm**
Currently, initial values to the numpyro and jax nuts samplers are automatically jittered. 

This PR exposes the jitter option and allows the boolean to be set through nuts_sampler_kwargs.

## Related Issue

There is no issue related to this change.
- [ ] Closes #
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7083.org.readthedocs.build/en/7083/

### CODE REVIEW

File: tests/sampling/test_jax.py [hunk 1]
```diff
@@ -154,6 +154,39 @@ def test_deterministic_samples(sampler):
     assert np.allclose(trace.posterior["b"].values, trace.posterior["a"].values / 2)
 
 
+@pytest.mark.parametrize(
+    "sampler",
+    [
+        # sample_blackjax_nuts,
```
Comment (VMBoehm) [inline tests/sampling/test_jax.py:160]: this test currently fails for sample_blackjack_nuts

`FAILED tests/sampling/test_jax.py::test_initvals_without_jitter[sample_blackjax_nuts] - TypeError: iteration over a 0-d array`

File: pymc/sampling/jax.py [hunk 2]
```diff
@@ -256,7 +256,7 @@ def _get_batched_jittered_initial_points(
     chains: int,
     initvals: Optional[Union[StartDict, Sequence[Optional[StartDict]]]],
     random_seed: RandomSeed,
-    jitter: bool = True,
+    jitter: Optional[bool] = True,
```
Comment (ricardoV94) [inline pymc/sampling/jax.py]: The `Optional` hint doesn't make sense. That's for when a variable can have a value of None, but here bool with True/False should suffice

File: pymc/sampling/jax.py [hunk 3]
```diff
@@ -564,6 +568,7 @@ def sample_numpyro_nuts(
     target_accept: float = 0.8,
     random_seed: Optional[RandomState] = None,
     initvals: Optional[Union[StartDict, Sequence[Optional[StartDict]]]] = None,
+    jitter: Optional[bool] = True,
```
Comment (ricardoV94) [inline pymc/sampling/jax.py]: Same here

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Thanks! Looks good. I just found an issue with the type hint changes.

File: pymc/sampling/jax.py (hunk 2 above)
Comment (VMBoehm) [reply pymc/sampling/jax.py]: Thanks @ricardoV94! Updated

File: pymc/sampling/jax.py (hunk 3 above)
Comment (VMBoehm) [reply pymc/sampling/jax.py]: Updated

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**